clusterloader2/api/validation/validation.go

```diff
@@ -0,0 +1,139 @@
+func VerifyConfig(c *api.Config) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if c.AutomanagedNamespaces < 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("automanagedNamespaces"), c.AutomanagedNamespaces, "must be non-negative"))
+	}
+	for i := range c.TuningSets {
+		allErrs = append(allErrs, verifyTuningSet(&c.TuningSets[i], field.NewPath("tuningSets").Index(i))...)
+	}
+	for i := range c.Steps {
+		allErrs = append(allErrs, verifyStep(&c.Steps[i], field.NewPath("steps").Index(i))...)
+	}
+	return allErrs
+}
+
+func verifyStep(s *api.Step, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for i := range s.Phases {
+		allErrs = append(allErrs, verifyPhase(&s.Phases[i], fldPath.Child("phases").Index(i))...)
+	}
+	for i := range s.Measurements {
+		allErrs = append(allErrs, verifyMeasurement(&s.Measurements[i], fldPath.Child("measurements").Index(i))...)
+	}
+	isPhasesEmpty := len(s.Phases) == 0
+	isMeasurementsEmpty := len(s.Measurements) == 0
+	if (isPhasesEmpty && isMeasurementsEmpty) || (!isPhasesEmpty && !isMeasurementsEmpty) {
+		allErrs = append(allErrs, field.Invalid(fldPath, "", "exactly one of phases and measurements should non-empty"))
+	}
+	return allErrs
+}
+
+func verifyPhase(p *api.Phase, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if p.NamespaceRange != nil {
+		allErrs = append(allErrs, verifyNamespaceRange(p.NamespaceRange, fldPath.Child("namespaceRange"))...)
+	}
+	if p.ReplicasPerNamespace < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("replicasPerNamespace"), p.ReplicasPerNamespace, "must be non-negative"))
+	}
+	for i := range p.ObjectBundle {
+		allErrs = append(allErrs, verifyObject(&p.ObjectBundle[i], fldPath.Child("objectBundle").Index(i))...)
+	}
+	return allErrs
+}
+
+func verifyObject(o *api.Object, fldPath *field.Path) field.ErrorList {
+	return verifyObjectType(&o.ObjectType, fldPath.Child("objectType"))
+}
+
+func verifyNamespaceRange(nr *api.NamespaceRange, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if nr.Min < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("min"), nr.Min, "must be non-negative"))
+	}
+	if nr.Max < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("max"), nr.Max, "must be non-negative"))
+	}
+	if nr.Max < nr.Min {
+		allErrs = append(allErrs, field.Invalid(fldPath, "", "max must be greater than min"))
+	}
+	return allErrs
+}
+
+func verifyObjectType(_ *api.ObjectType, _ *field.Path) field.ErrorList {
+	return nil
+}
+
+func verifyTuningSet(ts *api.TuningSet, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	tuningSetsNumber := 0
+	if ts.QpsLoad != nil {
+		allErrs = append(allErrs, verifyQpsLoad(ts.QpsLoad, fldPath.Child("qpsLoad"))...)
+		tuningSetsNumber++
+	}
+	if ts.RandomizedLoad != nil {
+		allErrs = append(allErrs, verifyRandomizedLoad(ts.RandomizedLoad, fldPath.Child("randomizedLoad"))...)
+		tuningSetsNumber++
```

Comment:
Instead of that, you should do (for consistency) the same what we're doing with validating VolumeSounrce for example:
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L428